crates/miden-objects/src/account/code/procedure.rs

```diff
@@ -149,6 +152,73 @@ impl Deserializable for AccountProcedureInfo {
     }
 }
 
+// PRINTABLE PROCEDURE
+// ================================================================================================
+
+/// A printable representation of a single account procedure.
+#[derive(Debug, Clone)]
+pub struct PrintableProcedure {
+    mast: Arc<MastForest>,
+    procedure_info: AccountProcedureInfo,
+    entrypoint: MastNode,
+}
+
+impl PrintableProcedure {
+    /// Creates a new PrintableProcedure instance from its components.
+    pub(crate) fn new(
+        mast: Arc<MastForest>,
+        procedure_info: AccountProcedureInfo,
+        entrypoint: MastNode,
+    ) -> Self {
+        Self { mast, procedure_info, entrypoint }
+    }
+
+    fn entrypoint(&self) -> &MastNode {
+        &self.entrypoint
+    }
+
+    fn storage_offset(&self) -> u8 {
+        self.procedure_info.storage_offset()
+    }
+
+    fn storage_size(&self) -> u8 {
+        self.procedure_info.storage_size()
+    }
+
+    fn mast_root(&self) -> &Digest {
+        self.procedure_info.mast_root()
+    }
+}
+
+impl PrettyPrint for PrintableProcedure {
+    fn render(&self) -> vm_core::prettier::Document {
+        use vm_core::prettier::*;
+        let procedure_root = self.mast_root();
+        let storage_offset = self.storage_offset();
+        let storage_size = self.storage_size();
+
+        indent(
+            0,
+            indent(
+                4,
+                text(format!("proc.{}", procedure_root))
+                    + nl()
+                    + text(format!("storage.{}.{}", storage_offset, storage_size))
+                    + nl()
+                    + indent(
+                        4,
+                        const_text("begin")
+                            + nl()
+                            + self.entrypoint().to_pretty_print(&self.mast).render(),
+                    )
+                    + nl()
+                    + const_text("end"),
+            ) + nl()
+                + const_text("end"),
+        )
```

Comment:
I would change this to print out only:
```
begin
    {procedure body}
end
```
But when printing the full account code, I'd still have the same output as you showed in https://github.com/0xPolygonMiden/miden-base/pull/1273#issuecomment-2807696213. This means that the `proc.{procedure_root}` wrapper should be handled in the `AccountCode` pretty-printer.

This way, we'd still print out the full account code as you have now, but we'd have the option to print only procedure code (w/o storage part) if we want to.

Reply:
I sent a commit with this change, delegates the proc.{} and storage to account code. `PrintableProcedure` now prints just the code.